pkg/strategies/type_number.go

```diff
@@ -2,18 +2,47 @@ package strategies
 
 import (
 	"fmt"
+	"reflect"
 )
 
 // TypeNumber is for numerical values:
 const TypeNumber = "NUMBER"
 
 // Number asserts the given parameters then passes on for evaluation:
 func Number(conditional string, options []interface{}, value interface{}) (bool, error) {
-
-	// Type assert the value:
-	assertedValue, ok := value.(float64)
-	if !ok {
-		return false, fmt.Errorf("Unable to assert value (%v) as float64", value)
+	var assertedValue float64
+	var ok bool
+
+	// Type switch on the value (because numbers can come in a bunch of interesting shapes and sizes):
+	switch value.(type) {
```

Comment:
You could say that this is fairly disgusting.

Reply:
yeah, but - I presume it is an idiomatic use when you wish to stick with just what the core libraries provide? Rather than using some decimal library? If so, I'm totally OK with it, and as you say, its what Javascript is doing.